src/System.Private.CoreLib/shared/System/String.Comparison.cs

```diff
@@ -56,6 +56,8 @@ private static bool EqualsOrdinalIgnoreCase(string strA, string strB)
 
             return CompareInfo.EqualsOrdinalIgnoreCase(ref strA.GetRawStringData(), ref strB.GetRawStringData(), strB.Length);
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static unsafe int CompareOrdinalHelper(string strA, string strB)
```

Comment:
This helper does not seem to be providing much value. Would it be better to just inline the call in the two places that this is caller, without all the Debug.Asserts that describe the custom contract?